### PR TITLE
fix(kafka): pass *args, **kwargs to TracedConsumer.__init__

### PR DESCRIPTION
**Pull request by rdghosal**
Pass arbitrary positional and keyword arguments to `TracedConsumer` so as to allow client to configure their instance of `KafkaConsumer`. This is, for example, [required for configuring a logger](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#kafka-client-configuration) internal to the Consumer, where a logger can only be passed as a kwarg.

Currently, passing such a kwarg leads to the following error:
```TypeError: TracedConsumer.__init__() got an unexpected keyword argument 'logger'```

Related to issue [#5837](https://github.com/DataDog/dd-trace-py/issues/5873)

---

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] OPTIONAL: PR description includes explicit acknowledgement of the performance implications of the change as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by Yun-Kim — COMMENTED** [review]
Hi @rdghosal, thanks for the fix! Let me know if you can add a regression test as well to make sure #5837 is fixed, otherwise I'm happy to add it for you 😄

**Comment by rdghosal** [comment]
Hi @Yun-Kim ! Thanks for reviewing this. Would it be possible to ask you to add these regression tests to the PR? I believe you should be able to tack on changes (let me know if not).

**Review by Yun-Kim — APPROVED** [review]
Thanks again for the contribution @rdghosal! 🥳

**Comment by rdghosal** [comment]
@Yun-Kim Thanks for adding that test! Will be sure to follow your example moving forward. Also, I see there are some failing checks. Please let me know if there's something I should contribute to get these passing.

File: tests/contrib/kafka/test_kafka.py [hunk 1]
```diff
@@ -63,6 +64,20 @@ def consumer(tracer):
     _consumer.close()
 
 
+def test_consumer_created_with_logger_does_not_raise(tracer):
+    """Test that adding a logger to a Consumer init does not raise any errors."""
+    logger = logging.getLogger()
```
Comment (mabdinur) [inline tests/contrib/kafka/test_kafka.py:69]:
```suggestion
    logger = logging.getLogger()
    # regression test for: DataDog/dd-trace-py/issues/5873
```

File: releasenotes/notes/pass-args-kwargs-to-kafka-f3ca8128a2c4d612.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+
```
Comment (mabdinur) [inline releasenotes/notes/pass-args-kwargs-to-kafka-f3ca8128a2c4d612.yaml:2]:
```suggestion
```

**Review by mabdinur — COMMENTED** [review]
left a few suggestions but aside from that LGTM

**Review by mabdinur — APPROVED** [review]

**Pull request merged**